### ISSUE TITLE
FatalError while using -H:-PartialEscapeAnalysis option

### ISSUE THREAD

**Issue by AntonHofer** [oracle/graal#3324]
**Describe the issue**
Invoking native-image with -H:-PartialEscapeAnalysis flag results in a RuntimeException. 
This happens regardless of the source code, the issue can be reproduced even with a simple "Hello world" program.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. Create a simple HelloWorld program
2. Invoke native-image with -H:-PartialEscapeAnalysis option
3. Observe the error

**Describe GraalVM and your environment:**
 - GraalVM version: **GraalVM CE 21.1.0-dev-20210331_1045**
 - JDK major version: **11**
 - OS: **Debian GNU/Linux bullseye/sid**
 - Architecture: **AMD64**

**More details**
```
[helloworld:30045]    classlist:   1,311.40 ms,  0.96 GB
[helloworld:30045]        (cap):     536.29 ms,  0.96 GB
[helloworld:30045]        setup:   2,307.62 ms,  0.96 GB
[helloworld:30045]     (clinit):     179.38 ms,  1.71 GB
[helloworld:30045]   (typeflow):   5,233.26 ms,  1.71 GB
[helloworld:30045]    (objects):   4,445.77 ms,  1.71 GB
[helloworld:30045]   (features):     456.31 ms,  1.71 GB
[helloworld:30045]     analysis:  10,527.79 ms,  1.71 GB
[helloworld:30045]     universe:     551.35 ms,  1.71 GB
[helloworld:30045]      (parse):   1,235.86 ms,  1.71 GB
[helloworld:30045]     (inline):   1,665.28 ms,  1.72 GB
[helloworld:30045]      compile:   3,008.01 ms,  1.72 GB
Fatal error:com.oracle.svm.core.util.VMError$HostedError: guarantee failed
        at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:68)
        at com.oracle.svm.core.util.VMError.guarantee(VMError.java:82)
        at com.oracle.svm.hosted.code.CompileQueue.removeDeoptTargetOptimizations(CompileQueue.java:1039)
        at com.oracle.svm.hosted.code.CompileQueue.createSuites(CompileQueue.java:383)
        at com.oracle.svm.hosted.code.CompileQueue.finish(CompileQueue.java:363)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:653)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$2(NativeImageGenerator.java:495)
        at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Error: Image build request failed with exit status 1
```

**Comment by oubidar-Abderrahim** [comment]
Thank you for reporting this issue, it is now tracked internally on GR-30564

**Comment by zapster** [comment]
The issue has been fixed in f910811dbbef1b243bdf0629e669b7193a9cd3a8.
